cmake/protobuf.cmake

```diff
@@ -90,6 +86,6 @@ elseif(gRPC_PROTOBUF_PROVIDER STREQUAL "package")
         set(_gRPC_PROTOBUF_PROTOC_EXECUTABLE ${PROTOBUF_PROTOC_EXECUTABLE})
       endif()
     endif()
-    set(_gRPC_FIND_PROTOBUF "if(NOT Protobuf_FOUND AND NOT PROTOBUF_FOUND)\n  find_package(Protobuf ${gRPC_PROTOBUF_PACKAGE_TYPE})\nendif()")
+    set(_gRPC_FIND_PROTOBUF "if(NOT Protobuf_FOUND AND NOT PROTOBUF_FOUND)\n  find_package(Protobuf CONFIG)\nendif()")
```

Comment:
Unrelated, but this is going into `gRPCConfig.cmake`, you should use `find_dependency()` and not `find_package()` in those:

https://cmake.org/cmake/help/latest/module/CMakeFindDependencyMacro.html
https://stackoverflow.com/questions/50263280/difference-between-find-dependency-and-find-package-in-cmake

Reply:
Oh that's a good point. Thanks! Updated this and will update others too.

Reply:
You need to `include(CMakeFindDependencyMacro)` to use `find_dependency`. #33361